fetch_from_gmail.py

```diff
@@ -230,43 +183,37 @@ def fetch_single_message(
         # used from the previous run. So this way, in the next run, after we paginate through all 100 emails of the
         # current page, and then on the next page, we encounter an email that we have already seen, then we will
         # can be sure that we have seen all emails in and before that page (assuming the BATCH SIZE has not been reduced compared to previous runs).
-        return "stop_early"  # Special return value to indicate continue
+        raise StopEarlyException()  # Special return value to indicate continue
```

Comment:
Minor nitpick, this is isn't a return anymore.

Reply:
fair.